examples/vello_editor/src/main.rs

```diff
@@ -114,6 +116,14 @@ impl ApplicationHandler for SimpleVelloApp<'_> {
         };
 
         self.editor.handle_event(event.clone());
+        if let Some(parley::Rect { x0, y0, x1, y1 }) = self.editor.preedit_area() {
+            println!("{x0} : {y0}");
+            println!("{x1} : {y1}");
+            render_state.window.set_ime_cursor_area(
+                LogicalPosition::new(x0, y0),
+                LogicalSize::new(x1 - x0, y1 - y0),
+            );
+        }
```

Comment:
There's discussion in #111 about when to notify the platform about the IME preedit area. @DJMcNab reported an infinite loop if the area is sent as a response to all IME events: https://github.com/linebender/parley/pull/111#issuecomment-2358707457. The simplest solution is to cache the previously sent area, and only send the current area if it's not the same. Unconditionally send the area on `IME::Enabled`.

To be a bit more conservative in reporting to the platform, the main thing missing is to easily know whether `Layout` has changed in response to an event, see also https://github.com/linebender/parley/pull/111#discussion_r1808975010.

Reply:
Almost all events affect either the cursor position or the Layout, so in the case of the example it doesn't much matter; however it might be good to make the calculation one-shot, maybe give it an observer pattern for ease of use.

Reply:
That is true, though some solution is required for the infinite loop that currently probably happens on some platforms (report area -> platform resends preedit -> report area -> …). Unfortunately I can't test this on my platform. `WindowEvent::{RedrawRequested,CursorMoved}` are possible exeptions.

Reply:
@tomcur I think #143 should make this pretty easy.